packages/service-provider-core/src/whatwg-url.ts

```diff
@@ -6,11 +6,21 @@ if (typeof require('util').TextDecoder !== 'function' || typeof require('util').
   Object.assign(require('util'), textEncodingPolyfill());
 }
 
-import { URL as WhatWGURL } from 'whatwg-url';
-import { URL as NodeURL } from 'url';
-export const URL = NodeURL ?? (WhatWGURL as typeof NodeURL);
+let URL: typeof import('url').URL;
 
-export function textEncodingPolyfill(): any {
+// Easiest way to get global `this` in any environment
```

Comment:
I think we should put some comments here about the general logic behind this if then else: like we want to use `window.URL` or `global.URL` if available, and we fall back to the whatwg-url for the Java shell and other environments.

Reply:
Added a more thorough description in 5e54ebe79c7af17b9b982846975b834d39ad44f4